File: src/test/java/com/github/qky666/selenidepom/test/java/mtp/pom/javapom/MainMenuWidget.java
package com.github.qky666.selenidepom.test.java.mtp.pom.javapom;

import com.codeborne.selenide.Condition;
import com.codeborne.selenide.SelenideElement;
import com.github.qky666.selenidepom.pom.Required;
import com.github.qky666.selenidepom.pom.Widget;
import lombok.Getter;

public class MainMenuWidget extends Widget {

    // Fields
    @Getter(lazy = true) @Required private final SelenideElement home = $("a.img-menu");
    @Getter(lazy = true) @Required private final SelenideElement services = $("li#servicios_menu>a");
    @Getter(lazy = true) @Required private final SelenideElement sectors = $("li#sectores_menu>a");
    @Getter(lazy = true) @Required private final SelenideElement training = $("li#formacion_menu>a");
    @Getter(lazy = true) @Required private final SelenideElement blog = $("li#blog_menu>a");
    @Getter(lazy = true) @Required private final SelenideElement talent = $("li#talento_menu>a");
    @Getter(lazy = true) @Required private final SelenideElement about = $("li.sobre_menu>a");
    @Getter(lazy = true) @Required private final SelenideElement contact = $("li#contacto_menu>a");
    @Getter(lazy = true) @Required private final SelenideElement langEn = $$("li.individual-menu-idioma>a").findBy(Condition.text("en"));
    @Getter(lazy = true) @Required private final SelenideElement langEs = $$("li.individual-menu-idioma>a").findBy(Condition.text("es"));
    @Getter(lazy = true) @Required private final SelenideElement openSearch = $("button#btn-menu");

    @Getter(lazy = true) private final SelenideElement servicesPopUp = $("div.dropdown-servicios");
    @Getter(lazy = true) private final SelenideElement servicesPopUpQualityAssurance = getServicesPopUp().$("a[data-principal='Aseguramiento de la calidad']");

    //Constructor
    public MainMenuWidget(SelenideElement self) {super(self);}
}

File: src/test/kotlin/com/github/qky666/selenidepom/test/kotlin/mtp/pom/menu/desktop/MainMenuWidget.kt
package com.github.qky666.selenidepom.test.kotlin.mtp.pom.menu.desktop

import com.codeborne.selenide.Condition.text
import com.codeborne.selenide.SelenideElement
import com.github.qky666.selenidepom.pom.Required
import com.github.qky666.selenidepom.pom.Widget

class MainMenuWidget(self: SelenideElement) : Widget(self) {
    @Required val services = find("li#servicios_menu>a")
    @Required val sectors = find("li#sectores_menu>a")
    @Required val training = find("li#formacion_menu>a")
    @Required val blog = find("li#blog_menu>a")
    @Required val talent = find("li#talento_menu>a")
    @Required val about = find("li.sobre_menu>a")
    @Required val contact = find("li#contacto_menu>a")
    @Required val langEn = findAll("li.individual-menu-idioma>a").findBy(text("en"))
    @Required val langEs = findAll("li.individual-menu-idioma>a").findBy(text("es"))

    // Search
    val searchOpen = find("button#btn-menu")
    val searchMenu = SearchMenuWidget(find("form#search-menu"))

    // Services Menu
    val servicesPopUp = ServicesPopupMenuWidget(find("div.dropdown-servicios"))
}